helm/client.go

```
@@ -164,25 +166,9 @@ func (c *Client) UpdateRelease(releaseName, chartName, chartVersion, namespace s
 
 	rel, err := client.Run(releaseName, ch, options)
 
-	// Forewords: Kubernetes API defines that some fields are immutable. In particular:
-	// For a StatefulSet, everything except spec.replicas, spec.template, and spec.updateStrategy (i.e spec.selector and spec.VolumeClaimTemplate is immutable)
-	// catch error "field is immutable" and than helm delete && helm install
```

I think this logic is not necessary when using --force with the upgrade command. Also, I would rather throw an error message in the UI than delete and re-install, since there are no known situations where --force should not work.

---

we had this situation during changing pod labels!

---

@sjmiller609 let's add more tests here to check all cases

---

wdyt to use the test framework I built already, but it's in python?

---

because these test cases are appropriate for functional rather than unit testing, right? is there somewhere appropriate for me to add a test for those cases right now?